rio/srcrobot/commands/TeleopSwerve.py

```diff
@@ -15,22 +15,39 @@ class TeleopSwerve(Command):
     strafeSup: Callable[[], float]
     rotationSup: Callable[[], float]
     robotCentricSup: Callable[[], bool]
+    slowSup: Callable[[], list[float]]
 
-    def __init__(self, s_Swerve, translationSup, strafeSup, rotationSup, robotCentricSup):
+    def __init__(self, s_Swerve, translationSup, strafeSup, rotationSup, robotCentricSup, slowSup=lambda: [0.0, 0.0]):
         self.s_Swerve = s_Swerve
         self.addRequirements(s_Swerve)
 
         self.translationSup = translationSup
         self.strafeSup = strafeSup
         self.rotationSup = rotationSup
         self.robotCentricSup = robotCentricSup
+        self.slowSup = slowSup
+
 
     def execute(self):
         # Get Values, Deadband
         translationVal = applyDeadband(self.translationSup(), Constants.stickDeadband)**3
         strafeVal = applyDeadband(self.strafeSup(), Constants.stickDeadband)**3
         rotationVal = self.getRotationValue()
 
+        # Apply slowmode
+        slowMove, slowTurn = self.slowSup()
+
+        # TODO: REMOVE THIS IN PRODUCTION. THIS IS TO SAVE THE ROBOT DURING TESTING.
+        if slowMove < 0 or slowTurn < 0:
+            print("SLOWMODE ERROR: SLOW OFFSET IS NEGATIVE")
+            print("OVERRIDING TO ZERO")
+            slowMove = 0
+            slowTurn = 0
+
+        translationVal -= translationVal*slowMove*Constants.Swerve.slowMoveModifier
+        strafeVal -= strafeVal*slowMove*Constants.Swerve.slowMoveModifier
```

Comment:
Maybe I am missing something here, but if the left trigger is not pulled at all, slowMove will be 0 and the resulting translation and strafe values will be 0. Similarly for rotation. Do we need to reverse the trigger values? Some think like the following (the triggers we are using give [0...1]):

trigger = abs(trigger - 1.0)

Reply:
You might be missing the `-` in the `-=` statement. The calculation on the right is *subtracted* from the normal speed. If the trigger isn't held down at all, then we don't want to engage slow mode, and so we shouldn't subtract anything from the movement speed unless it's held down.

If it is held down, then we want the slow move modifier to come into full effect (being 0.5 here, the calculation would work out to `translation = translation - (translation*1*0.5)`, and for any value in-between, would slow it down slightly less (`translation = translation - (translation*0.5*0.5)` for halfway down, etc.)

And yes, the triggers return 0 when not held and 1 when held, at least when using the `getXXXTriggerAxis()` functions,

Reply:
Yes, that is what I was missing.